### ISSUE TITLE
Flux configuration should allow for non-ssh auth for git

### ISSUE THREAD

**Issue by pjlittle** [microsoft/bedrock#940]
Last week, we encountered a scenario where we were unable to use SSH keys to authenticate with git.  HTTPS has become a more common way to access git (e.g. https://<token>@github.com/owner/repo.git) and it's possible that there are customers who would prefer to authenticate this way.  

ask: Bedrock tooling supports HTTPS, in addition to SSH, for git authentication

**Comment by andrebriggs** [comment]
Hi @pjlittle, are you referring to taking advantage of [this](https://docs.fluxcd.io/en/latest/guides/use-git-https.html) Flux configuration?

**Comment by pjlittle** [comment]
Yes - I believe I found similar documentation on GitHub.  Being able to specify/pipe a git url via HTTPS and use a token to authenticate would be great. Every now and again we find that some customers have policies that require them to use HTTPS. Would be great to have the option to configure Flux that way, via Bedrock.
